lib/matplotlib/tests/test_widgets.py

```diff
@@ -328,12 +383,13 @@ def onselect(epress, erelease):
     xdata_new, ydata_new = xdata + xdiff, ydata + ydiff
     click_and_drag(tool, start=(xdata, ydata), end=(xdata_new, ydata_new),
                    key=use_key)
-    assert tool.extents == (extents[0] + ydiff, extents[1],
-                            ydata_new, extents[3])
+    assert_allclose(tool.extents, (xdata_new, extents[1],
+                                   extents[2] + xdiff, extents[3]))
```

Comment:
This the result of fixing a bug in resizing a square selector. Previously:

![Screen Recording 2022-01-08 at 15 12 53](https://user-images.githubusercontent.com/6197628/148649438-4c0e213d-9241-4521-b7a7-31241a3b59fd.gif)

This is now fixed such that when resizing a square selector the edge of the selector always lies on on the cursor.